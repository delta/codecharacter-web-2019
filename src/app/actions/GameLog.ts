import { action } from 'typesafe-actions';

export namespace GameLogActions {
  export enum Type {
    UPDATE_GAME_LOGS = 'UPDATE_GAME_LOGS',
    RESET_GAME_LOG_STATE = 'RESET_GAME_LOG_STATE',
    UPDATE_DISPLAY_DEBUG_LOG = 'UPDATE_DISPLAY_DEBUG_LOG',
    CLEAR_DISPLAY_DEBUG_LOG = 'CLEAR_DISPLAY_DEBUG_LOG',
  }

  export const updateGameLog = (
    player1DebugLog: string,
    player2DebugLog: string,
    gameLog: string,
  ) =>
    action(Type.UPDATE_GAME_LOGS, {
      gameLog,
      player1DebugLog,
      player2DebugLog,
    });

  export const resetGameLogState = () => action(Type.RESET_GAME_LOG_STATE);

  export const updateDisplayDebugLog = (log: string) =>
    action(Type.UPDATE_DISPLAY_DEBUG_LOG, { log });

  export const clearDisplayDebugLog = () => action(Type.CLEAR_DISPLAY_DEBUG_LOG);
}
