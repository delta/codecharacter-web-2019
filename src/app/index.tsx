import Dashboard from 'app/containers/Dashboard';
// @ts-ignore
import { initializeRendererAssets } from 'codecharacter-renderer';
import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router';

initializeRendererAssets();

/* tslint:disable-next-line:variable-name */
export const App = hot(module)(() => (
  <Switch>
    <Route path="/" component={Dashboard} />
  </Switch>
));
