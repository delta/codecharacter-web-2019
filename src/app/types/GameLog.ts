import { GameLogActions } from 'app/actions/GameLog';
import { ActionType } from 'typesafe-actions';

export interface StateProps {
  debugLog: string;
}

export interface ElementProps {
  height: number;
}

export type Props = StateProps & ElementProps;

const actions = {
  clearAllLogs: GameLogActions.clearAllLogs,
  clearDisplayDebugLog: GameLogActions.clearDisplayDebugLog,
  resetGameLog: GameLogActions.resetGameLogState,
  updateDisplayDebugLog: GameLogActions.updateDisplayDebugLog,
  updateGameLog: GameLogActions.updateGameLog,
  updateMatchPlayerId: GameLogActions.updateMatchPlayerId,
};

export interface GameLogStoreState {
  displayDebugLog: string;
  gameLog: string;
  player1DebugLog: string;
  player2DebugLog: string;
  matchPlayerId: number;
}

export type GameLogStoreAction = ActionType<typeof actions>;
