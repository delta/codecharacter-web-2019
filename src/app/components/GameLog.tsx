import * as GameLogInterfaces from 'app/types/GameLog';
import * as React from 'react';
// tslint:disable-next-line:import-name
import AceEditor from 'react-ace';

import 'brace/mode/c_cpp';
import 'brace/theme/terminal';

export class GameLog extends React.Component<GameLogInterfaces.Props, {}> {
  constructor(props: GameLogInterfaces.Props) {
    super(props);
  }

  public render() {
    const options = {
      enableBasicAutocompletion: false,
      enableLiveAutocompletion: false,
      enableSnippets: false,
      showLineNumbers: true,
      tabSize: 2,
    };

    return (
      <AceEditor
        mode="c_cpp"
        theme={'terminal'}
        name="game_log_div"
        fontSize={14}
        wrapEnabled={true}
        readOnly={true}
        showPrintMargin={false}
        showGutter={true}
        highlightActiveLine={false}
        setOptions={options}
        editorProps={{ $blockScrolling: true }}
        width={'100%'}
        height={'100%'}
        value={this.props.debugLog}
      />
    );
  }
}
