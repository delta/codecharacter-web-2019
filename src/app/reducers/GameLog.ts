import { GameLogActions } from 'app/actions';
import * as GameLogInterfaces from 'app/types/GameLog';

const gameLogStoreIntialState: GameLogInterfaces.GameLogStoreState = {
  displayDebugLog: '',
  gameLog: '',
  matchPlayerId: 1,
  player1DebugLog: '',
  player2DebugLog: '',
};

export const gameLogReducer = (
  state = gameLogStoreIntialState,
  action: GameLogInterfaces.GameLogStoreAction,
) => {
  switch (action.type) {
    case GameLogActions.Type.UPDATE_GAME_LOGS: {
      return {
        ...state,
        gameLog: action.payload.gameLog,
        player1DebugLog: action.payload.player1DebugLog,
        player2DebugLog: action.payload.player2DebugLog,
      };
    }
    case GameLogActions.Type.RESET_GAME_LOG_STATE: {
      return {
        ...gameLogStoreIntialState,
      };
    }
    case GameLogActions.Type.UPDATE_DISPLAY_DEBUG_LOG: {
      return {
        ...state,
        displayDebugLog: `${state.displayDebugLog}${action.payload.log}`,
      };
    }
    case GameLogActions.Type.CLEAR_DISPLAY_DEBUG_LOG: {
      return {
        ...state,
        displayDebugLog: '',
      };
    }
    case GameLogActions.Type.UPDATE_MATCH_PLAYER_ID: {
      return {
        ...state,
        matchPlayerId: action.payload.matchPlayerId,
      };
    }
    default:
      return state;
  }
};
