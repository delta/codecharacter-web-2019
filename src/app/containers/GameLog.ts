import { GameLog } from 'app/components/GameLog';
import { RootState } from 'app/reducers';
import * as GameLogInterfaces from 'app/types/GameLog';
import { connect } from 'react-redux';

const mapStateToProps = (rootState: RootState) => {
  return {
    debugLog: rootState.gameLog.displayDebugLog,
  };
};

const gameLogContainer = connect<GameLogInterfaces.StateProps, {}, {}>(mapStateToProps)(GameLog);

export default gameLogContainer;
