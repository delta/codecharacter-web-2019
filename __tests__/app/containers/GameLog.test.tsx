import GameLog from 'app/containers/GameLog';
import { configureStore } from 'app/store';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('GameLog Container', () => {
  const { store } = configureStore();
  const wrapper = shallow(<GameLog height={100} />, {
    context: {
      store,
    },
  });

  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have prop', () => {
    expect(wrapper.props().debugLog).toMatch(store.getState().gameLog.displayDebugLog);
  });
});
