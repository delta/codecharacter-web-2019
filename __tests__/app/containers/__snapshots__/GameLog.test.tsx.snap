// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameLog Container Should render 1`] = `
<GameLog
  debugLog=""
  dispatch={[Function]}
  height={100}
/>
`;
